cypress/e2e/functional/tile_tests/geometry_tool_spec.js

```diff
@@ -277,6 +356,164 @@ context('Geometry Tool', function () {
     geometryToolTile.getGraphPoint().should("have.length", 0);
     geometryToolTile.getSelectedGraphPoint().should("have.length", 0);
 
+    // Create first polygon from existing points
+    cy.log('Create first polygon from existing points');
+    clueCanvas.clickToolbarButton('geometry', 'point');
+    geometryToolTile.clickGraphPosition(0, 0);
+    geometryToolTile.clickGraphPosition(10, 0);
+    geometryToolTile.clickGraphPosition(5, 5);
+    clueCanvas.clickToolbarButton('geometry', 'polygon');
+    geometryToolTile.getGraphPoint().should("have.length", 3);
+    geometryToolTile.getGraphPoint().eq(0).click();
+    geometryToolTile.getGraphPoint().eq(1).click();
+    geometryToolTile.getGraphPoint().eq(2).click();
+    geometryToolTile.getGraphPoint().eq(0).click();
+    geometryToolTile.getGraphPolygon().should("have.length", 1);
+    geometryToolTile.getGraphPoint().should("have.length", 3);
+
+    // Add a point to the existing polygon
+    cy.log('Add a point to the existing polygon');
+    geometryToolTile.clickGraphPosition(10, 0); // Reuse existing point
+    geometryToolTile.clickGraphPosition(15, 5);
+    geometryToolTile.clickGraphPosition(5, 5); // Reuse existing point
+    clueCanvas.clickToolbarButton('geometry', 'polygon');
+    // check number of points
+    geometryToolTile.getGraphPoint().should("have.length", 4);
+
+    // Verify the polygon count is still 1
+    geometryToolTile.getGraphPolygon().should("have.length", 1);
+
+    // Create a second polygon that shares the same points as the first
+    cy.log('Create a second polygon that shares the same points as the first');
+    clueCanvas.clickToolbarButton('geometry', 'polygon');
+    geometryToolTile.clickGraphPosition(15, 10); // new point
+    geometryToolTile.clickGraphPosition(15, 5); // shared point
+    geometryToolTile.clickGraphPosition(20, 5); // new point
+    geometryToolTile.clickGraphPosition(15, 10); // close the polygon
+
+    // Point should be shared
+    geometryToolTile.getGraphPoint().should("have.length", 6); // New point added
+
+    // Store the original coordinates for comparison
+    let originalCx, originalCy;
+    clueCanvas.clickToolbarButton('geometry', 'select');
+    geometryToolTile.clickGraphPosition(15, 5); // shared point
+    geometryToolTile.getSelectedGraphPoint().then(($point) => {
+      originalCx = parseFloat($point.attr('cx'));
+      originalCy = parseFloat($point.attr('cy'));
+      cy.wrap(originalCx).as('originalCx');
+      cy.wrap(originalCy).as('originalCy');
```

Comment:
For a moment, this confused me: I thought that "cy.wrap(..." was referring to 'cy'. the point's y value, rather than to 'cypress'. Perhaps the 'cx' & 'cy' variable names should be changed to something like 'px' & 'py' to avoid this issue? I'm not sure what the 'c' refers to, anyway. (I guess, to be consistent, you would also need to change 'originalCx' & 'originalCy' to 'originalPx' & 'originalPy'.)